File: src/test/java/TestsKT3.py
from selenium import webdriver
import pytest
from ProductPage import ProductPage

class TestsKT3:
    @pytest.fixture(scope="session")
    def product_page(self, driver):
        return ProductPage(driver)


    @pytest.mark.parametrize("product_index", [0, 1, 2])
    def test_add_to_wishlist(self, product_page, product_index):
        product_page.open_main_page()
        product_page.click_product_name(product_index)
        product_page.find_and_click_element(product_page.WISHLIST_ON_PRODUCT_PAGE_BUTTON)
        assert product_page.is_login_alert_displayed()

    @pytest.mark.parametrize("expected", [" Success: You have added Canon EOS 5D to your shopping cart! "])
    def test_add_camera(self, product_page, expected):
        product_page.open_main_page()
        product_page.open_cameras_page()
        product_page.click_product_name(product_index=0)
        product_page.open_optional_options()
        product_page.select_red_color()
        product_page.find_and_click_element(product_page.ADD_TO_CART_BUTTON)

        if product_page.is_cart_alert_displayed():
            if product_page.is_this_cart_alert(expected):
                assert True
            else:
                print("Это уведомление не о добавлении в корзину")
                assert False
        else:
            print("Уведомление не появилось")
            assert False

    @pytest.mark.parametrize("expected", [" Success: You have added Samsung Galaxy Tab 10.1 to your shopping cart! "])
    def test_add_tablet(self, product_page, expected):
        product_page.open_main_page()
        product_page.open_tablets_page()
        product_page.click_product_name(product_index=0)
        product_page.find_and_click_element(product_page.ADD_TO_CART_BUTTON)

        if product_page.is_cart_alert_displayed():
            if product_page.is_this_cart_alert(expected):
                assert True
            else:
                print("This is a wrong alert")
                assert False
        else:
            print("There was no alert")
            assert False

    @pytest.mark.parametrize("expected", [" Success: You have added HTC Touch HD to your shopping cart! "])
    def test_add_htc_phone(self, product_page, expected):
        product_page.open_main_page()
        product_page.open_phones_page()
        product_page.click_product_name(product_index=0)
        product_page.find_and_click_element(product_page.ADD_TO_CART_BUTTON)

        if product_page.is_cart_alert_displayed():
            if product_page.is_this_cart_alert(expected):
                assert True
            else:
                print("This is a wrong alert")
                assert False
        else:
            print("There was no alert")
            assert False

    @pytest.mark.parametrize("product_index", [0])
    @pytest.mark.parametrize("name", ["me"])
    @pytest.mark.parametrize("review_text", ["finally some really good stuff"])
    @pytest.mark.parametrize("rating", [5])
    def test_write_review(self, driver, product_page, product_index, name, review_text, rating):
        product_page.open_main_page()
        product_page.click_product_name(product_index)
        product_page.find_and_click_element(product_page.REVIEWS_BUTTON)
        product_page.write_a_review(name, review_text)
        product_page.rate_the_product(rating)
        product_page.find_and_click_element(product_page.CONTINUE_REVIEW_BUTTON)
        actual_reviewer_name = product_page.get_actual_reviewer_name()
        actual_review_text = product_page.get_actual_review_text()
        assert actual_reviewer_name == name and actual_review_text == review_text

@pytest.fixture(scope="session")
def driver():
    driver = webdriver.Firefox()
    driver.maximize_window()
    yield driver
    driver.quit()

File: src/test/java/TestsKT3.java
import io.qameta.allure.Description;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import io.qameta.allure.Story;
import org.junit.jupiter.api.*;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;
import org.junit.jupiter.params.provider.ValueSource;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import static org.junit.jupiter.api.Assertions.*;

class TestsKT3 {

    private static WebDriver driver;
    private static ProductPage productPage;

    @BeforeAll
    static void setUp() {
        // Инициализация драйвера
        driver = new FirefoxDriver();
        driver.manage().window().maximize();
        productPage = new ProductPage(driver);
    }

    @AfterAll
    static void tearDown() {
        // Закрытие драйвера после всех тестов
        if (driver != null) {
            driver.quit();
        }
    }

    @ParameterizedTest
    @ValueSource(ints = {0, 1, 2})
    @Story("Добавление в вишлист")
    @Description("Проверка добавления товара в список желаний")
    @Severity(SeverityLevel.CRITICAL)
    @DisplayName("Добавление товара в вишлист")
    void testAddToWishlist(int productIndex) {
        productPage.openMainPage();
        productPage.clickProductName(productIndex, null, 3);
        productPage.findAndClickElement(ProductPage.WISHLIST_ON_PRODUCT_PAGE_BUTTON, null, 3);
        assertTrue(productPage.isLoginAlertDisplayed());
    }

    @ParameterizedTest
    @ValueSource(strings = {" Success: You have added Canon EOS 5D to your shopping cart! "})
    void testAddCamera(String expected) throws InterruptedException {
        productPage.openMainPage();
        productPage.openCamerasPage();
        productPage.clickProductName(0, null, 3);
        productPage.openOptionalOptions();
        productPage.selectRedColor();
        productPage.findAndClickElement(ProductPage.ADD_TO_CART_BUTTON, null, 3);

        if (productPage.isCartAlertDisplayed()) {
            if (productPage.isThisCartAlert(expected)) {
                assertTrue(true);
            } else {
                System.out.println("This is a wrong alert");
                assertTrue(false);
            }
        } else {
            System.out.println("There was no alert");
            assertTrue(false);
        }
    }

    @ParameterizedTest
    @ValueSource(strings = {" Success: You have added Samsung Galaxy Tab 10.1 to your shopping cart! "})
    void testAddTablet(String expected) {
        productPage.openMainPage();
        productPage.openTabletsPage();
        productPage.clickProductName(0, null, 3);
        productPage.findAndClickElement(ProductPage.ADD_TO_CART_BUTTON, null, 3);

        if (productPage.isCartAlertDisplayed()) {
            if (productPage.isThisCartAlert(expected)) {
                assertTrue(true);
            } else {
                System.out.println("This is a wrong alert");
                assertTrue(false);
            }
        } else {
            System.out.println("There was no alert");
            assertTrue(false);
        }
    }

    @ParameterizedTest
    @ValueSource(strings = {" Success: You have added HTC Touch HD to your shopping cart! "})
    void testAddHtcPhone(String expected) {
        productPage.openMainPage();
        productPage.openPhonesPage();
        productPage.clickProductName(0, null, 3);
        productPage.findAndClickElement(ProductPage.ADD_TO_CART_BUTTON, null, 3);

        if (productPage.isCartAlertDisplayed()) {
            if (productPage.isThisCartAlert(expected)) {
                assertTrue(true);
            } else {
                System.out.println("This is a wrong alert");
                assertTrue(false);
            }
        } else {
            System.out.println("There was no alert");
            assertTrue(false);
        }
    }

    @ParameterizedTest
    @CsvSource({
            "0, me, hello, 5"
    })
    void testWriteReview(int productIndex, String name, String reviewText, int rating) throws InterruptedException {
        productPage.openMainPage();
        productPage.clickProductName(productIndex, null, 3);
        productPage.findAndClickElement(ProductPage.REVIEWS_BUTTON, null, 3);
        productPage.writeAReview(name, reviewText);
        productPage.rateTheProduct(rating);
        productPage.findAndClickElement(ProductPage.CONTINUE_REVIEW_BUTTON, null, 3);

        String actualReviewerName = productPage.getActualReviewerName();
        String actualReviewText = productPage.getActualReviewText();

        assertEquals(name, actualReviewerName);
        assertEquals(reviewText, actualReviewText);
    }
}